src/ExpressionEvaluator/Core/Source/ExpressionCompiler/LanguageInstructionDecoder.cs

```diff
@@ -38,16 +38,24 @@ string IDkmLanguageInstructionDecoder.GetMethodName(DkmLanguageInstructionAddres
                 // but it was ignored.  Furthermore, it's not clear what FullNames would mean with respect
                 // to argument names in C# or Visual Basic.  For consistency with the old behavior, we'll
                 // just ignore the flag as well.
-                Debug.Assert((argumentFlags & (DkmVariableInfoFlags.FullNames | DkmVariableInfoFlags.Names | DkmVariableInfoFlags.Types)) == argumentFlags,
+                Debug.Assert((argumentFlags & (DkmVariableInfoFlags.FullNames | DkmVariableInfoFlags.Names | DkmVariableInfoFlags.Types | DkmVariableInfoFlags.CompactName)) == argumentFlags,
```

Comment:
Could we be stricter than this? 
1. It looks like `DkmVariableInfoFlags.CompactName` should only be used by itself, is that right?
2. If so, we could throw (instead of merely asserting) if it's used in conjunction with other flags (since we'd ignore them anyways).

Reply:
Also, it looks like we don't have any direct tests on this method. Is that something we could cover?

Reply:
We could be stricter and throw an exception if certain values are used in combination with `DkmVariableInfoFlags.CompactName`. But there might be reasons in the future to combine `CompactName` with other values, including existing values, that we'd disallow as a result.